src/TypeSpec.Extension/Emitter.Csharp/src/lib/confidentLevels.ts

```diff
@@ -0,0 +1,78 @@
+import {
+    InputDictionaryType,
+    InputListType,
+    InputModelType,
+    InputType,
+    isInputDictionaryType,
+    isInputListType,
+    isInputModelType
+} from "../type/inputType.js";
+
+export function getConfident(
+    type: InputType,
+    visitedModels: Set<InputModelType>
+): boolean {
+    if (isInputModelType(type)) {
+        return getModelConfident(type, visitedModels);
+    } else if (isInputListType(type)) {
+        return getListConfident(type, visitedModels);
+    } else if (isInputDictionaryType(type)) {
+        return getDictionaryConfident(type, visitedModels);
+    }
+
+    return type.IsConfident ?? true; // by default they are confident
+}
+
+function getModelConfident(
+    type: InputModelType,
+    visitedModels: Set<InputModelType>
+): boolean {
+    if (visitedModels.has(type)) return true; // this means this has been calculated before or being calculated right now, return
```

Comment:
```suggestion
    if (visitedModels.has(type)) return type.IsConfident; // this means this has been calculated before or being calculated right now, return
```

Reply:
And we don't need `visitedModels` anymore. `type.IsConfident` itself is already a cache.

Reply:
> And we don't need `visitedModels` anymore. `type.IsConfident` itself is already a cache.

this is not true, we need that cache in a case of a model referencing itself.

Reply:
> 

this I agree